cms/djangoapps/contentstore/views/item.py

```diff
@@ -852,6 +852,11 @@ def safe_get_username(user_id):
         visibility_state = None
     published = modulestore().has_published_version(xblock) if not is_library_block else None
 
+    #instead of adding a new feature directly into xblock-info, we should add them into override_type.
+    override_type = {}
+    if getattr(xblock, "is_entrance_exam", None):
+        override_type['is_entrance_exam'] = xblock.is_entrance_exam
```

Comment:
@mattdrayer instead of adding "is_entrance_exam" into xblock info directly, i put it in a override_type dict. so every new feature we can utilize that dict. this way we can get the xblock type. i hope it will address the @andy-armstrong feedback